Source/Diagnostics/WarpXOpenPMD.H

```diff
@@ -0,0 +1,78 @@
+#ifndef WARPX_OPEN_PMD_H_
+#define WARPX_OPEN_PMD_H_
+
+#include <MultiParticleContainer.H> // has AMReX Vector etc used below
+//#include <WarpX.H>
+
+#include <openPMD/openPMD.hpp>
+
+class WarpXOpenPMDParticle
+{
+public:
+  WarpXOpenPMDParticle(bool, const std::string& filetype);
+  //WarpXOpenPMDParticle(const std::string& dir, const std::string& fileType);
+  ~WarpXOpenPMDParticle();
+
+  void SetStep(int ts);
+  void SetOpenPMDType(const std::string& type);
+
+  void SaveContainerPlots(const std::unique_ptr<MultiParticleContainer>&);
+
+  void WriteOpenPMDFields(//const std::string& filename,
```

Comment:
yes,  originally I wrote it for particles. Then during our zoom meeting I found out the need to include field in the same file. Will change it to avoid future confusion.

Reply:
fields can go in the same series, but if I remember correctly we concluded to just implement it together or separate, just what is easier.
we do have some field output already merged in the plotfile routines, which I would like to do separately since it's easier to control that way.

Reply:
There is one series in this class.  Both field and particle use that series. 
There. are two functions,  WriteOpenPMDFields() for field data,  and SaveContainerPlots() for particle data.  WriteOpenPMDFields is from the existing implementation in WarpX.  I. just moved it in this class so all data can use the. same series.

Reply:
Ok, let us then just rename them for clarity to something like `WriteOpenPMDFields` and `WriteOpenPMDParticles`. No reason to keep the historic names then, they will just confuse us down the road.